### PR TITLE
Use arrow funcs as the default test name

### PR DESCRIPTION
**Pull request by gsnedders**
[Rendered](https://github.com/web-platform-tests/rfcs/blob/test_arrow_name/rfcs/test_arrow_name.md)

see also https://github.com/web-platform-tests/wpt/pull/25853

### CODE REVIEW

**Comment by jgraham** [comment]
I think this is a good idea. My main concern is that different browsers may stringify the functions differently and end up with different test names. Do we have any insight into how likely that is?

**Comment by gsnedders** [comment]
> Do we have any insight into how likely that is?

@othermaciej said they were "pretty interoperable in practice (or at least good enough for this purpose)"

**Review by jgraham — APPROVED** [review]
I suppose if we run into known dfferences we can think of a way to discourage people from using the problematic constructs (or get the differences fixed)

**Review by stephenmcgruer — APPROVED** [review]
LGTM. I really appreciate that this finds a non-invasive way to address a concern that some folks have about testharness.js. It is a concern I don't personally agree with, but I love that you were able to address it (hopefully to their satisfaction) without impacting other users :)

File: rfcs/test_arrow_name.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Details
+
+At the moment, we practically require all tests to have a name be explicitly given. This makes testharness.js tests much more verbose than WebKit/Blink's js-test.js. For example, currently in WebKit we have:
+
+```javascript
+const shadowHost = document.createElement("div");
+const shadowRoot = shadowHost.attachShadow({mode: 'closed'});
+const shadowRootChild = shadowRoot.appendChild(document.createElement('div'));
+
+shouldBeFalse('shadowHost.isConnected');
```
[the remark this answers is not in the document]
Comment (gsnedders) [reply rfcs/test_arrow_name.md:16]: nononononononononononononononononononononononononononononononononononono, just say no

File: rfcs/test_arrow_name.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# RFC: Use simple arrow funcs as the default test name
+
+## Summary
+
+For `test(() => assert_true(true))`, use `assert_true(true)` as the default test name rather than `document.title`
+
+## Details
+
+At the moment, we practically require all tests to have a name be explicitly given. This makes testharness.js tests much more verbose than WebKit/Blink's js-test.js. For example, currently in WebKit we have:
```
[the remark this answers is not in the document]
Comment (gsnedders) [reply rfcs/test_arrow_name.md:9]: (There was some before your time. I don't really know what made it go away, though.)

**Comment by Hexcles** [comment]
Just want to confirm that this won't apply to single-page tests as they don't meet the criteria of "simple arrow func", right?

**Comment by jgraham** [comment]
It won't because they don't. I don't know what we could do for single page tests, given that the use case there is typically more complex event-based things.

**Comment by zcorpan** [comment]
Single page tests can continue to use the title for the test name.

About how interoperable `Function.prototype.toString` is. Here are test results for test262 for JSC, V8, SpinderMonkey:

https://test262.report/browse/built-ins/Function/prototype/toString?engines=javascriptcore%2Cspidermonkey%2Cv8

I think the only relevant test is `arrow-function.js`, which passes in those 3 JS engines. The test doesn't test handling of newlines, but [testing manually](http://software.hixie.ch/utilities/js/live-dom-viewer/saved/8541) it seems consistent.

File: rfcs/test_arrow_name.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# RFC: Use simple arrow funcs as the default test name
```
Comment (stephenmcgruer) [inline rfcs/test_arrow_name.md:1]:
```suggestion
# RFC 64: Use simple arrow funcs as the default test name
```

**Comment by othermaciej** [comment]
> > Do we have any insight into how likely that is?
> 
> @othermaciej said they were "pretty interoperable in practice (or at least good enough for this purpose)"

Adding to this, the latest ECMAScript spec [defines what happens pretty exactly](https://tc39.es/ecma262/#sec-function.prototype.tostring) for functions that have source text.

The current spec language comes from [this proposal](https://tc39.es/Function-prototype-toString-revision/), which explicitly lists the goal:

* For functions defined using ECMAScript code, toString must return source text slice from beginning of first token to end of last token matched by the appropriate grammar production

So I think it's safe to count on interoperability here.

**Comment by gsnedders** [comment]
Yay, nobody objected! 🎉

**Pull request merged**